cs/src/Management/IdGeneration.cs

```diff
@@ -0,0 +1,26 @@
+    private static readonly Random random = new Random();
```

Comment:
`Random` is not thread safe. Use `Random.Shared`.

Reply:
Random.Shared is not allowed since we also compile to netcore. I created a separate thread safe random to use here